src/thing/Config.cpp

```diff
@@ -0,0 +1,48 @@
+#include "Arduino.h"
+#include "thing/Config.h"
+#include "third-party/arduino-json-5.6.7/include/ArduinoJson.h"
+
+namespace thing {
+
+void Config::SerializeToJson(Stream* output, std::function<void(int size)> handle_size) const {
+  DynamicJsonBuffer jsonBuffer;
+  JsonObject& root = jsonBuffer.createObject();
+  root["host"] = host.c_str();
+  root["auth"] = auth.c_str();
+  root["path"] = path.c_str();
+  root["wifi_ssid"] = wifi_ssid.c_str();
+  root["wifi_key"] = wifi_key.c_str();
+  root["analog_activation"] = analog_activation_threshold;
+  root["wifi_connect_attempts"] = wifi_connect_attempts;
+
+  JsonObject& pins_root = root.createNestedObject("pins");
+  pins_root["digital_in"] = pins.digital_in;
+  pins_root["digital_out"] = pins.digital_out;
+  pins_root["analog_in"] = pins.analog_in;
+  pins_root["analog_out"] = pins.analog_out;
+  pins_root["config_mode_button"] = pins.config_mode_button;
+
+  handle_size(root.measureLength());
+  root.printTo(*output);
+}
+
+void Config::ReadFromJson(const char* string) {
```

Comment:
IIRC when you pass a const string to ArduinoJson it will duplicate and copy the string in the Buffer memory:
https://github.com/bblanchon/ArduinoJson/blob/master/include/ArduinoJson/JsonBuffer.hpp#L87